plugins/aws/appmesh/appmeshVGHealthChecks.js

```diff
@@ -0,0 +1,97 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'App Mesh VG Health Check Policies',
+    category: 'App Mesh',
+    domain: 'Reliability',
+    description: 'Ensure that Amazon App Mesh virtual gateways use health check policies to monitor the availability of virtual nodes.',
+    more_info: 'Health check policies in App Mesh are essential to maintain application availability and reliability by monitoring the health of associated virtual nodes.',
+    link: 'https://docs.aws.amazon.com/app-mesh/latest/userguide/virtual_gateway_health_checks.html',
+    recommended_action: 'Configure health check policies for the virtual gateway listeners in your App Mesh, specifying values for healthy threshold, health check interval, health check protocol, timeout period, and unhealthy threshold.',
+    apis: ['AppMesh:listMeshes', 'AppMesh:listVirtualGateways', 'AppMesh:describeVirtualGateway'],
+   
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var region = helpers.regions(settings);
+
+        async.each(region.appmesh, function(region, rcb) {
+            var listMeshes = helpers.addSource(cache, source,
+                ['appmesh', 'listMeshes', region]);
+
+            if (!listMeshes) return rcb();
+
+            if (listMeshes.err || !listMeshes.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query for App Mesh meshes: ' + helpers.addError(listMeshes), region);
+                return rcb();
+            }
+
+            if (!listMeshes.data.length) {
+                helpers.addResult(results, 0, 'No App Mesh meshes found', region);
+                return rcb();
+            }
+
+            for (let mesh of listMeshes.data){
+                if (!mesh.arn || !mesh.meshName) continue;
+
+                let resource = mesh.arn;
+
+                var listVirtualGateways = helpers.addSource(cache, source,
+                    ['appmesh', 'listVirtualGateways', region, mesh.meshName]);
+
+                if (!listVirtualGateways || listVirtualGateways.err || !listVirtualGateways.data) {
+                    helpers.addResult(results, 3,
+                        'Unable to query for App Mesh virtual gateways: ' + helpers.addError(listVirtualGateways),
+                        region, resource);
+                    continue;
+                }
+
+                if (!listVirtualGateways.data.virtualGateways) {
+                    helpers.addResult(results, 0,
+                        'No App Mesh virtual gateways found', region, resource);
+                    continue;
+                }
```

Comment:
```suggestion
                if (!listVirtualGateways.data.virtualGateways || !listVirtualGateways.data.virtualGateways.length) {
                    helpers.addResult(results, 0,
                        'No App Mesh virtual gateways found', region, resource);
                    continue;
                }
```

virtualGateways has an empty array if there are no gateways